doc/3.Algorithms.md

```diff
@@ -238,13 +238,13 @@ Assertion 3.5.8:
 
 Assertion 3.5.9:
     SpdmMessage.BaseAsymSel only has one bit at most.
-    if (CHAL_CAP == 1 || MEAS_CAP == 2) then (BaseAsymSel == one of {TPM_ALG_RSASSA_2048, TPM_ALG_RSAPSS_2048, TPM_ALG_RSASSA_3072, TPM_ALG_RSAPSS_3072, TPM_ALG_ECDSA_ECC_NIST_P256, TPM_ALG_RSASSA_4096, TPM_ALG_RSAPSS_4096, TPM_ALG_ECDSA_ECC_NIST_P384, TPM_ALG_ECDSA_ECC_NIST_P521})
-    if (CHAL_CAP == 0 && MEAS_CAP != 2) then (BaseAsymSel == 0)
+    if (CERT_CAP == 1 || CHAL_CAP == 1 || MEAS_CAP == 2 || KEY_EX_CAP == 1) then (BaseAsymSel == one of {TPM_ALG_RSASSA_2048, TPM_ALG_RSAPSS_2048, TPM_ALG_RSASSA_3072, TPM_ALG_RSAPSS_3072, TPM_ALG_ECDSA_ECC_NIST_P256, TPM_ALG_RSASSA_4096, TPM_ALG_RSAPSS_4096, TPM_ALG_ECDSA_ECC_NIST_P384, TPM_ALG_ECDSA_ECC_NIST_P521})

```

Comment:
Why PUB_KEY_ID_CAP requires HASH ?

Reply:
`PUB_KEY_ID_CAP` requires `BaseAsymSel`.

Reply:
PUB_KEY_ID_CAP just means how to provision the pub key. I think the usage is already covered - CHAL_CAP == 1 || MEAS_CAP == 2 || KEY_EX_CAP == 1.

Do you see a case that PUB_KEY_ID_CAP == 1, but CHAL_CAP == 0 && MEAS_CAP != 2 && KEY_EX_CAP == 0 ?

Reply:
The same can be said for `CERT_CAP == 1`. 
>Do you see a case that PUB_KEY_ID_CAP == 1, but CHAL_CAP == 0 && MEAS_CAP != 2 && KEY_EX_CAP == 0 ?

No, but the specification does not prohibit it. If the validation test (and specification) want to prohibit that then we should put it in the specification.

Reply:
"does not prohibit" != "shall be set".

We can only test "shall". We cannot test "not prohibit".

Reply:
I'll file an issue against the specification and we can see what the SPDM WG says tomorrow.

Reply:
That is fine. My point is:
The conformance test shall not test what is NOT described in the spec.
The conformance test shall only test the sentence with "shall" in the spec.

Since the spec does not say how to use "PUB_KEY_ID_CAP" (It only says: the pub key is provisioned), we shall not mandate BaseAsym to be 1 or 0.

Reply:
That's fine but if you're going to leave `PUB_KEY_ID_CAP` out then you should also leave `CERT_CAP` out.

Reply:
CERT_CAP is different. We need include CERT_CAP, because the pub key in cert shall match the BaseAsym. We shall test. 

PUB_KEY_ID_CAP cannot return anything. Nothing to test.

Reply:
According to @t-koulouris's [recent clarification in the 1.3 specification](https://github.com/DMTF/SPDM-WG/pull/2072) `BaseAsymSel` is `0` in the case where `(CERT_CAP == 1 && CHAL_CAP == 0 && MEAS_CAP != 2 && KEY_EX_CAP == 0)`, assuming that's a legal combination. I suppose in 1.1 and 1.2 it is more vague.

Reply:
Sorry, I did not get your point.
Would you please express your opinion clearly?